types/index.d.ts

```diff
@@ -322,6 +331,8 @@ export interface ChannelListProps {
     e: Client.Event,
   ): void;
   setActiveChannelOnMount?: boolean;
+  /** Optional function to filter channels prior to loading in the DOM */
+  filterFunction?: <T>(channels: T) => T;
```

Comment:
this should not be a generic IMO, what is passed in is an array of Immturable Channels and the return should also be the same type